ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,301 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/if.hpp"
+#include <algorithm>
+#include <iterator>
+#include <ngraph/validation_util.hpp>
+#include "itt.hpp"
+#include "ngraph/factory.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/util/multi_subgraph_base.hpp"
+#include "ngraph/specialize_function.hpp"
+
+#include "ngraph/runtime/reference/if.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(ngraph::op::v8::If, "If", 8);
+
+op::v8::If::If()
+    : MultiSubGraphOp(2)
+{
+}
+
+op::v8::If::If(const Output<Node>& execution_condition)
+    : If()
+{
+    set_argument(0, execution_condition);
+}
+
+ngraph::PartialShape resolve_shape(const ngraph::PartialShape& then_pshape,
```

Comment:
please add description for this function. why is it needed?

Reply:
Also, I think it make sense to add this function into anonymous namepspace to make it static.
Otherwise, there might be some names conflicts during linking in future.